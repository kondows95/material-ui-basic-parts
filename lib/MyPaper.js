"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var core_1 = require("@material-ui/core");
var styles_1 = require("@material-ui/core/styles");
var useStyles = styles_1.makeStyles(function (theme) { return ({
    paper: {
        margin: theme.spacing(2, 1, 0),
        padding: theme.spacing(2),
        display: 'grid',
    },
}); });
var MyPaper = function (props) {
    var classes = useStyles();
    return react_1.default.createElement(core_1.Paper, { className: classes.paper }, props.children);
};
exports.default = MyPaper;
