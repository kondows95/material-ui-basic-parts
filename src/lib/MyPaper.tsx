import React, { PropsWithChildren } from 'react';
import { Paper, Theme } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) => ({
    paper: {
        margin: theme.spacing(2, 1, 0), //TXB,
        padding: theme.spacing(2),
        display: 'grid',
    },
}));

const MyPaper: React.FC<PropsWithChildren<{}>> = (props) => {
    const classes = useStyles();
    return <Paper className={classes.paper}>{props.children}</Paper>;
};

export default MyPaper;
