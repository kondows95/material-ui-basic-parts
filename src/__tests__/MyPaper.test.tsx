import React from 'react';
import TestLib from 'react-testing-library-wrapper';
import MyPaper from '../lib/MyPaper';

describe('Basic Usage', () => {
    let tLib: TestLib;
    beforeEach(() => {
        tLib = new TestLib(
            (
                <MyPaper data-testid="myPaper">
                    <div>MyChild</div>
                </MyPaper>
            )
        );
    });

    it('You can add data-testid', () => {
        tLib.get('myPaper');
    });

    it('You can add chiled component', () => {
        expect(tLib.render.container.textContent).toBe('MyChild');
    });

    it('Snapshot', () => {
        expect(tLib.render.asFragment()).toMatchSnapshot();
    });
});
