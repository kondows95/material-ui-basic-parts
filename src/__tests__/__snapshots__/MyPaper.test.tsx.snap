// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Usage Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root makeStyles-paper-30 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div>
      MyChild
    </div>
  </div>
</DocumentFragment>
`;
